Strange Error ECONNRESET

Hi guys!

today i have a very strange error, look this schema.prisma.

```prisma
datasource db {
  provider = "sqlite"
  url      = "file:dev.db"
}

generator photon {
  provider = "/home/neo/golib/bin/photongo"
  output   = "./photon/photon_gen.go"
  package  = "photon"
}

model User {
    id       String   @default(cuid()) @id @unique
    name     Boolean
    password String
    rol      Rol[]
}

model Rol {
    id          String    @default(cuid()) @id @unique
    create      DateTime  @default(now())
    update      DateTime  @updatedAt
    name        String
    users       User[]
    permissions Permission[]
}

model Permission {
    id     String    @default(cuid()) @id @unique
    create DateTime  @default(now())
    key    String
    rol    Rol[]
}

```
Works very well, pass lift migrations without problems, generate the Photon Go client, but if you want add another field in any model for example in model **User** i just add the field **email** and looks like:

```prisma
model User {
    id       String   @default(cuid()) @id @unique
    name     Boolean
    password String
    email    String
    rol      Rol[]
}

```
:boom: i get this error:

```bash
go run github.com/prisma/photongo generate
> Downloading Prisma engines for debian-openssl-1.1.x [====================] 100%

Generating Photon Go to ./../photon.go
/__enclose_io_memfs__/build/index.js:217148
        throw ex;
        ^

Error: read ECONNRESET
    at _errnoException (util.js:1022:11)
    at Pipe.onread (net.js:610:25)
panic: could not run [generate]: exit status 7

goroutine 1 [running]:
main.main()
        /home/neo/golib/pkg/mod/github.com/prisma/photongo@v0.0.0-20191217175644-fc2a3926061a/main.go:19 +0x15a
exit status 2
```

Any ideas??? thanks! :hugs:

--- a/generator.go
+++ b/generator.go
@@ -5,10 +5,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"io"
-	"log"
 	"os"
-	"os/signal"
-	"syscall"
 
 	"github.com/prisma/photongo/generator"
 	"github.com/prisma/photongo/jsonrpc"
@@ -30,25 +27,19 @@ func reply(w io.Writer, data interface{}) error {
 	return nil
 }
 
-func invokePrisma() {
-	// make sure to exit when signal triggers
-	c := make(chan os.Signal)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+func invokePrisma() error {
+	reader := bufio.NewReader(os.Stdin)
 
-	go func() {
-		<-c
-		os.Exit(1)
-	}()
-
-	scanner := bufio.NewScanner(os.Stdin)
-	for scanner.Scan() {
-		content := scanner.Bytes()
+	for {
+		content, err := reader.ReadBytes('\n')
+		if err != nil {
+			return fmt.Errorf("could not read bytes from stdin: %w", err)
+		}
 
 		var input jsonrpc.Request
 
-		err := json.Unmarshal(content, &input)
-		if err != nil {
-			log.Fatalf("could not open stdin %s", err)
+		if err := json.Unmarshal(content, &input); err != nil {
+			return fmt.Errorf("could n		ot open stdin %s", err)
 		}
 
 		var response interface{}
@@ -71,18 +62,18 @@ func invokePrisma() {
 			var params generator.Root
 
 			if err := json.Unmarshal(input.Params, &params); err != nil {
-				log.Fatalf("could not unmarshal params into generator.Root type %s", err)
+				return fmt.Errorf("could not unmarshal params into generator.Root type %s", err)
 			}
 
 			if err = generator.Run(&params); err != nil {
-				log.Fatalf("could not generate code. %s", err)
+				return fmt.Errorf("could not generate code. %s", err)
 			}
+		default:
+			return fmt.Errorf("no such method %s", input.Method)
 		}
 
-		err = reply(os.Stderr, jsonrpc.NewResponse(input.ID, response))
-
-		if err != nil {
-			log.Fatalf("could not open stdin %s", err)
+		if err := reply(os.Stderr, jsonrpc.NewResponse(input.ID, response)); err != nil {
+			return fmt.Errorf("could not open stdin %s", err)
 		}
 	}
 }

--- a/main.go
+++ b/main.go
@@ -3,7 +3,10 @@ package main
 //go:generate go build .
 
 import (
+	"log"
 	"os"
+	"os/signal"
+	"syscall"
 
 	"github.com/prisma/photongo/cli"
 	"github.com/prisma/photongo/logger"
@@ -14,13 +17,30 @@ func main() {
 		args := os.Args[1:]
 		logger.L.Printf("invoking command %+v", args)
 		// prisma CLI
-		err := cli.Run(args, true)
-		if err != nil {
+		if err := cli.Run(args, true); err != nil {
 			panic(err)
 		}
-	} else {
-		logger.L.Printf("invoking prisma")
-		// invoke the prisma generator
-		invokePrisma()
+
+		return
 	}
+
+	// running the prisma generator
+
+	logger.L.Printf("invoking prisma")
+
+	// exit when signal triggers
+	c := make(chan os.Signal)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		<-c
+		os.Exit(1)
+	}()
+
+	if err := invokePrisma(); err != nil {
+		log.Printf("error occurred when invoking prisma: %s", err)
+		os.Exit(1)
+	}
+
+	logger.L.Printf("success")
 }

